src/java.desktop/share/classes/javax/swing/JViewport.java

```diff
@@ -614,7 +615,11 @@ private Graphics getBackingStoreGraphics(Graphics g) {
     private void paintViaBackingStore(Graphics g) {
         Graphics bsg = getBackingStoreGraphics(g);
         try {
-            super.paint(bsg);
+            if (g instanceof SunGraphics2D) {
```

Comment:
Can we move the check in `getBackingStoreGraphics` which does not use the passed parameter currently
instead of these 2 places?